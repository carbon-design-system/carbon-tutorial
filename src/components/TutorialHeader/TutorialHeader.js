import React from 'react';
import Notification20 from '@carbon/icons-react/lib/notification/20';
import UserAvatar20 from '@carbon/icons-react/lib/user--avatar/20';
import AppSwitcher20 from '@carbon/icons-react/lib/app-switcher/20';
import {
  Header,
  HeaderName,
  HeaderNavigation,
  HeaderMenuItem,
  HeaderGlobalBar,
  HeaderGlobalAction,
  SkipToContent,
} from 'carbon-components-react/lib/components/UIShell';
import { Link } from 'react-router-dom';

const TutorialHeader = () => (
  <Header aria-label="Carbon Tutorial">
    <SkipToContent />
    <HeaderName element={Link} to="/" prefix="IBM">
      Carbon Tutorial
    </HeaderName>
    <HeaderNavigation aria-label="Carbon Tutorial">
      <HeaderMenuItem element={Link} to="/repos">
        Repositories
      </HeaderMenuItem>
    </HeaderNavigation>
    <HeaderGlobalBar>
      <HeaderGlobalAction aria-label="Notifications">
        <Notification20 />
      </HeaderGlobalAction>
      <HeaderGlobalAction aria-label="User Avatar">
        <UserAvatar20 />
      </HeaderGlobalAction>
      <HeaderGlobalAction aria-label="App Switcher">
        <AppSwitcher20 />
      </HeaderGlobalAction>
    </HeaderGlobalBar>
  </Header>
);

export default TutorialHeader;
