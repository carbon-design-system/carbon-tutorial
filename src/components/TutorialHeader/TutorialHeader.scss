/*
 * TutorialHeader css File
*/

.TutorialHeader {
  /* propertys here */
}
