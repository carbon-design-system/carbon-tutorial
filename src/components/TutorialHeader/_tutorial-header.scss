@use '@carbon/react/scss/colors';

.cds--header__global .cds--popover-content {
  background-color: colors.$gray-20;
}

.cds--header__global .cds--popover-caret {
  background-color: colors.$gray-20;
}
