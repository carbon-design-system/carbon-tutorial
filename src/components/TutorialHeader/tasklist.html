<!DOCTYPE html>
<html dir="ltr" lang="en-US">
<head>
	<meta charset="UTF-8" />
	<title></title>
	<style>
	body {font-family: helvetica; line-height: 1.4}
	#root div {width: 50%; float: left;}
	li {list-style: none; position: relative;}
	li:not(.done):hover {cursor: pointer; color: #0088cc;}
	li::before {content: "\2192"; position: absolute; top: 0; left: -1.3em;}
	.done {text-decoration: line-through; color: #aaa;}
	.done::before {content: "\2713"; text-decoration: none; left: -1em;}
	</style>

</head>

<body>
	<div id="root">
		<div>
			<h2>all tasks</h2>
			<ul>
				<li v-for="task in tasks" v-text="task.description" :class="{ done: task.completed  }" @click="toggleState(task)"></li>
			</ul>
		</div>
		<div>
			<h2>to do</h2>
			<ul>
				<li v-for="task in incompleteTasks" v-text="task.description"></li>
			</ul>
			<h2>done</h2>
			<ul>
				<li v-for="task in completeTasks" v-text="task.description" class="done"></li>
			</ul>
		</div>
	</div>

	<script src="https://unpkg.com/vue@2.5.2/dist/vue.js"></script>

	<script>
		var app = new Vue({
			el: '#root',
			data: {
				tasks : [
					{ description: 'go to the store', completed: true} ,
					{ description: 'howl at the moon', completed: false} ,
					{ description: 'feed the cat', completed: true} ,
					{ description: 'make dinner', completed: false} ,
					{ description: 'wash dishes', completed: true}
				]
			},
			computed: {
				incompleteTasks(){
					return this.tasks.filter(task => ! task.completed);
				},
				completeTasks(){
					return this.tasks.filter(task => task.completed);
				}
			},
			methods: {
				toggleState(task){
					task.completed = true;
				}
			}
		});

	</script>

</body>
</html>
