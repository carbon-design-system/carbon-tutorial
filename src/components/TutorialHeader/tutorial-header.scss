header {
  &.bx--header {
    li {
      list-style: none;
    }
  }
}
