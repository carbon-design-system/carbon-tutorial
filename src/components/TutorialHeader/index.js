import TutorialHeader from './Header';
export default TutorialHeader;
