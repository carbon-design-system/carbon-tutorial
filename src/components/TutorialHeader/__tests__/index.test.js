// test file
import React from 'react';
import { shallow } from 'enzyme';
import TutorialHeader from '../TutorialHeader';

describe('Header', () => {
  it('exists', () => {
    const wrapper = shallow(<TutorialHeader />);
    expect(wrapper).toMatchSnapshot();
  });
});
