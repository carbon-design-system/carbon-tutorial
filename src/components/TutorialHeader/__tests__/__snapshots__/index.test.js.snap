// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header exists 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TutorialHeader />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Header
        aria-label="Carbon Tutorial"
      >
        <SkipToContent
          href="#main-content"
          tabIndex="0"
        >
          Skip to main content
        </SkipToContent>
        <HeaderName
          href="/"
          prefix="IBM"
        >
          Carbon Tutorial
        </HeaderName>
        <HeaderNavigation
          aria-label="Carbon Tutorial"
        >
          <ForwardRef
            href="/repos"
          >
            Link 1
          </ForwardRef>
        </HeaderNavigation>
        <HeaderGlobalBar>
          <HeaderGlobalAction>
            <ForwardRef
              height={20}
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 32 32"
              width={20}
              xmlns="http://www.w3.org/2000/svg"
            />
          </HeaderGlobalAction>
          <HeaderGlobalAction>
            <ForwardRef
              height={20}
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 32 32"
              width={20}
              xmlns="http://www.w3.org/2000/svg"
            />
          </HeaderGlobalAction>
          <HeaderGlobalAction>
            <ForwardRef
              height={20}
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 20 20"
              width={20}
              xmlns="http://www.w3.org/2000/svg"
            />
          </HeaderGlobalAction>
        </HeaderGlobalBar>
      </Header>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "aria-label": "Carbon Tutorial",
        "children": Array [
          <SkipToContent
            href="#main-content"
            tabIndex="0"
          >
            Skip to main content
          </SkipToContent>,
          <HeaderName
            href="/"
            prefix="IBM"
          >
            Carbon Tutorial
          </HeaderName>,
          <HeaderNavigation
            aria-label="Carbon Tutorial"
          >
            <ForwardRef
              href="/repos"
            >
              Link 1
            </ForwardRef>
          </HeaderNavigation>,
          <HeaderGlobalBar>
            <HeaderGlobalAction>
              <ForwardRef
                height={20}
                preserveAspectRatio="xMidYMid meet"
                viewBox="0 0 32 32"
                width={20}
                xmlns="http://www.w3.org/2000/svg"
              />
            </HeaderGlobalAction>
            <HeaderGlobalAction>
              <ForwardRef
                height={20}
                preserveAspectRatio="xMidYMid meet"
                viewBox="0 0 32 32"
                width={20}
                xmlns="http://www.w3.org/2000/svg"
              />
            </HeaderGlobalAction>
            <HeaderGlobalAction>
              <ForwardRef
                height={20}
                preserveAspectRatio="xMidYMid meet"
                viewBox="0 0 20 20"
                width={20}
                xmlns="http://www.w3.org/2000/svg"
              />
            </HeaderGlobalAction>
          </HeaderGlobalBar>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Skip to main content",
            "href": "#main-content",
            "tabIndex": "0",
          },
          "ref": null,
          "rendered": "Skip to main content",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Carbon Tutorial",
            "href": "/",
            "prefix": "IBM",
          },
          "ref": null,
          "rendered": "Carbon Tutorial",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "aria-label": "Carbon Tutorial",
            "children": <ForwardRef
              href="/repos"
            >
              Link 1
            </ForwardRef>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Link 1",
              "href": "/repos",
            },
            "ref": null,
            "rendered": "Link 1",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "propTypes": Object {
                "children": [Function],
                "className": [Function],
                "element": [Function],
                "role": [Function],
              },
              "render": [Function],
            },
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <HeaderGlobalAction>
                <ForwardRef
                  height={20}
                  preserveAspectRatio="xMidYMid meet"
                  viewBox="0 0 32 32"
                  width={20}
                  xmlns="http://www.w3.org/2000/svg"
                />
              </HeaderGlobalAction>,
              <HeaderGlobalAction>
                <ForwardRef
                  height={20}
                  preserveAspectRatio="xMidYMid meet"
                  viewBox="0 0 32 32"
                  width={20}
                  xmlns="http://www.w3.org/2000/svg"
                />
              </HeaderGlobalAction>,
              <HeaderGlobalAction>
                <ForwardRef
                  height={20}
                  preserveAspectRatio="xMidYMid meet"
                  viewBox="0 0 20 20"
                  width={20}
                  xmlns="http://www.w3.org/2000/svg"
                />
              </HeaderGlobalAction>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <ForwardRef
                  height={20}
                  preserveAspectRatio="xMidYMid meet"
                  viewBox="0 0 32 32"
                  width={20}
                  xmlns="http://www.w3.org/2000/svg"
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "height": 20,
                  "preserveAspectRatio": "xMidYMid meet",
                  "viewBox": "0 0 32 32",
                  "width": 20,
                  "xmlns": "http://www.w3.org/2000/svg",
                },
                "ref": null,
                "rendered": null,
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "defaultProps": Object {
                    "height": 20,
                    "preserveAspectRatio": "xMidYMid meet",
                    "viewBox": "0 0 32 32",
                    "width": 20,
                    "xmlns": "http://www.w3.org/2000/svg",
                  },
                  "displayName": "Notification20",
                  "propTypes": Object {
                    "aria-hidden": [Function],
                    "aria-label": [Function],
                    "aria-labelledby": [Function],
                    "children": [Function],
                    "className": [Function],
                    "height": [Function],
                    "preserveAspectRatio": [Function],
                    "tabIndex": [Function],
                    "viewBox": [Function],
                    "width": [Function],
                    "xmlns": [Function],
                  },
                  "render": [Function],
                },
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <ForwardRef
                  height={20}
                  preserveAspectRatio="xMidYMid meet"
                  viewBox="0 0 32 32"
                  width={20}
                  xmlns="http://www.w3.org/2000/svg"
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "height": 20,
                  "preserveAspectRatio": "xMidYMid meet",
                  "viewBox": "0 0 32 32",
                  "width": 20,
                  "xmlns": "http://www.w3.org/2000/svg",
                },
                "ref": null,
                "rendered": null,
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "defaultProps": Object {
                    "height": 20,
                    "preserveAspectRatio": "xMidYMid meet",
                    "viewBox": "0 0 32 32",
                    "width": 20,
                    "xmlns": "http://www.w3.org/2000/svg",
                  },
                  "displayName": "UserAvatar20",
                  "propTypes": Object {
                    "aria-hidden": [Function],
                    "aria-label": [Function],
                    "aria-labelledby": [Function],
                    "children": [Function],
                    "className": [Function],
                    "height": [Function],
                    "preserveAspectRatio": [Function],
                    "tabIndex": [Function],
                    "viewBox": [Function],
                    "width": [Function],
                    "xmlns": [Function],
                  },
                  "render": [Function],
                },
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <ForwardRef
                  height={20}
                  preserveAspectRatio="xMidYMid meet"
                  viewBox="0 0 20 20"
                  width={20}
                  xmlns="http://www.w3.org/2000/svg"
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "height": 20,
                  "preserveAspectRatio": "xMidYMid meet",
                  "viewBox": "0 0 20 20",
                  "width": 20,
                  "xmlns": "http://www.w3.org/2000/svg",
                },
                "ref": null,
                "rendered": null,
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "defaultProps": Object {
                    "height": 20,
                    "preserveAspectRatio": "xMidYMid meet",
                    "viewBox": "0 0 20 20",
                    "width": 20,
                    "xmlns": "http://www.w3.org/2000/svg",
                  },
                  "displayName": "AppSwitcher20",
                  "propTypes": Object {
                    "aria-hidden": [Function],
                    "aria-label": [Function],
                    "aria-labelledby": [Function],
                    "children": [Function],
                    "className": [Function],
                    "height": [Function],
                    "preserveAspectRatio": [Function],
                    "tabIndex": [Function],
                    "viewBox": [Function],
                    "width": [Function],
                    "xmlns": [Function],
                  },
                  "render": [Function],
                },
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Header
          aria-label="Carbon Tutorial"
        >
          <SkipToContent
            href="#main-content"
            tabIndex="0"
          >
            Skip to main content
          </SkipToContent>
          <HeaderName
            href="/"
            prefix="IBM"
          >
            Carbon Tutorial
          </HeaderName>
          <HeaderNavigation
            aria-label="Carbon Tutorial"
          >
            <ForwardRef
              href="/repos"
            >
              Link 1
            </ForwardRef>
          </HeaderNavigation>
          <HeaderGlobalBar>
            <HeaderGlobalAction>
              <ForwardRef
                height={20}
                preserveAspectRatio="xMidYMid meet"
                viewBox="0 0 32 32"
                width={20}
                xmlns="http://www.w3.org/2000/svg"
              />
            </HeaderGlobalAction>
            <HeaderGlobalAction>
              <ForwardRef
                height={20}
                preserveAspectRatio="xMidYMid meet"
                viewBox="0 0 32 32"
                width={20}
                xmlns="http://www.w3.org/2000/svg"
              />
            </HeaderGlobalAction>
            <HeaderGlobalAction>
              <ForwardRef
                height={20}
                preserveAspectRatio="xMidYMid meet"
                viewBox="0 0 20 20"
                width={20}
                xmlns="http://www.w3.org/2000/svg"
              />
            </HeaderGlobalAction>
          </HeaderGlobalBar>
        </Header>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "aria-label": "Carbon Tutorial",
          "children": Array [
            <SkipToContent
              href="#main-content"
              tabIndex="0"
            >
              Skip to main content
            </SkipToContent>,
            <HeaderName
              href="/"
              prefix="IBM"
            >
              Carbon Tutorial
            </HeaderName>,
            <HeaderNavigation
              aria-label="Carbon Tutorial"
            >
              <ForwardRef
                href="/repos"
              >
                Link 1
              </ForwardRef>
            </HeaderNavigation>,
            <HeaderGlobalBar>
              <HeaderGlobalAction>
                <ForwardRef
                  height={20}
                  preserveAspectRatio="xMidYMid meet"
                  viewBox="0 0 32 32"
                  width={20}
                  xmlns="http://www.w3.org/2000/svg"
                />
              </HeaderGlobalAction>
              <HeaderGlobalAction>
                <ForwardRef
                  height={20}
                  preserveAspectRatio="xMidYMid meet"
                  viewBox="0 0 32 32"
                  width={20}
                  xmlns="http://www.w3.org/2000/svg"
                />
              </HeaderGlobalAction>
              <HeaderGlobalAction>
                <ForwardRef
                  height={20}
                  preserveAspectRatio="xMidYMid meet"
                  viewBox="0 0 20 20"
                  width={20}
                  xmlns="http://www.w3.org/2000/svg"
                />
              </HeaderGlobalAction>
            </HeaderGlobalBar>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Skip to main content",
              "href": "#main-content",
              "tabIndex": "0",
            },
            "ref": null,
            "rendered": "Skip to main content",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Carbon Tutorial",
              "href": "/",
              "prefix": "IBM",
            },
            "ref": null,
            "rendered": "Carbon Tutorial",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "aria-label": "Carbon Tutorial",
              "children": <ForwardRef
                href="/repos"
              >
                Link 1
              </ForwardRef>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Link 1",
                "href": "/repos",
              },
              "ref": null,
              "rendered": "Link 1",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "propTypes": Object {
                  "children": [Function],
                  "className": [Function],
                  "element": [Function],
                  "role": [Function],
                },
                "render": [Function],
              },
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <HeaderGlobalAction>
                  <ForwardRef
                    height={20}
                    preserveAspectRatio="xMidYMid meet"
                    viewBox="0 0 32 32"
                    width={20}
                    xmlns="http://www.w3.org/2000/svg"
                  />
                </HeaderGlobalAction>,
                <HeaderGlobalAction>
                  <ForwardRef
                    height={20}
                    preserveAspectRatio="xMidYMid meet"
                    viewBox="0 0 32 32"
                    width={20}
                    xmlns="http://www.w3.org/2000/svg"
                  />
                </HeaderGlobalAction>,
                <HeaderGlobalAction>
                  <ForwardRef
                    height={20}
                    preserveAspectRatio="xMidYMid meet"
                    viewBox="0 0 20 20"
                    width={20}
                    xmlns="http://www.w3.org/2000/svg"
                  />
                </HeaderGlobalAction>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <ForwardRef
                    height={20}
                    preserveAspectRatio="xMidYMid meet"
                    viewBox="0 0 32 32"
                    width={20}
                    xmlns="http://www.w3.org/2000/svg"
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "height": 20,
                    "preserveAspectRatio": "xMidYMid meet",
                    "viewBox": "0 0 32 32",
                    "width": 20,
                    "xmlns": "http://www.w3.org/2000/svg",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "defaultProps": Object {
                      "height": 20,
                      "preserveAspectRatio": "xMidYMid meet",
                      "viewBox": "0 0 32 32",
                      "width": 20,
                      "xmlns": "http://www.w3.org/2000/svg",
                    },
                    "displayName": "Notification20",
                    "propTypes": Object {
                      "aria-hidden": [Function],
                      "aria-label": [Function],
                      "aria-labelledby": [Function],
                      "children": [Function],
                      "className": [Function],
                      "height": [Function],
                      "preserveAspectRatio": [Function],
                      "tabIndex": [Function],
                      "viewBox": [Function],
                      "width": [Function],
                      "xmlns": [Function],
                    },
                    "render": [Function],
                  },
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <ForwardRef
                    height={20}
                    preserveAspectRatio="xMidYMid meet"
                    viewBox="0 0 32 32"
                    width={20}
                    xmlns="http://www.w3.org/2000/svg"
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "height": 20,
                    "preserveAspectRatio": "xMidYMid meet",
                    "viewBox": "0 0 32 32",
                    "width": 20,
                    "xmlns": "http://www.w3.org/2000/svg",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "defaultProps": Object {
                      "height": 20,
                      "preserveAspectRatio": "xMidYMid meet",
                      "viewBox": "0 0 32 32",
                      "width": 20,
                      "xmlns": "http://www.w3.org/2000/svg",
                    },
                    "displayName": "UserAvatar20",
                    "propTypes": Object {
                      "aria-hidden": [Function],
                      "aria-label": [Function],
                      "aria-labelledby": [Function],
                      "children": [Function],
                      "className": [Function],
                      "height": [Function],
                      "preserveAspectRatio": [Function],
                      "tabIndex": [Function],
                      "viewBox": [Function],
                      "width": [Function],
                      "xmlns": [Function],
                    },
                    "render": [Function],
                  },
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <ForwardRef
                    height={20}
                    preserveAspectRatio="xMidYMid meet"
                    viewBox="0 0 20 20"
                    width={20}
                    xmlns="http://www.w3.org/2000/svg"
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "height": 20,
                    "preserveAspectRatio": "xMidYMid meet",
                    "viewBox": "0 0 20 20",
                    "width": 20,
                    "xmlns": "http://www.w3.org/2000/svg",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "defaultProps": Object {
                      "height": 20,
                      "preserveAspectRatio": "xMidYMid meet",
                      "viewBox": "0 0 20 20",
                      "width": 20,
                      "xmlns": "http://www.w3.org/2000/svg",
                    },
                    "displayName": "AppSwitcher20",
                    "propTypes": Object {
                      "aria-hidden": [Function],
                      "aria-label": [Function],
                      "aria-labelledby": [Function],
                      "children": [Function],
                      "className": [Function],
                      "height": [Function],
                      "preserveAspectRatio": [Function],
                      "tabIndex": [Function],
                      "viewBox": [Function],
                      "width": [Function],
                      "xmlns": [Function],
                    },
                    "render": [Function],
                  },
                },
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
